charms/kserve-controller/src/charm.py

```diff
@@ -256,7 +271,25 @@ def _controller_pebble_layer(self):
                             "SECRET_NAME": "kserve-webhook-server-cert",
                         },
                     },
-                }
+                },
+                "checks": {
```

Comment:
suggestion: implement the `on-check-failure` for the 2 checks added here, this is how pebble knows what to do to the pebble service when the check fails. Otherwise, the check is not really affecting the container status IIUC.
[Reference](https://documentation.ubuntu.com/pebble/reference/health-checks/)

Reply:
Ahh good catch!

Pushed a commit e429d6f